src/layers/base-layer.ts

```diff
@@ -55,16 +57,94 @@ import {getSampleData} from 'utils/table-utils/data-container-utils';
 
 import {hexToRgb, getColorGroupByName, reverseColorRange} from 'utils/color-utils';
 
-/** @typedef {import('./index').Layer} LayerClass} */
+import {RGBColor, RGBAColor, MapState, Filter, Datasets} from 'reducers';
+import {LayerTextLabel, ColorUI} from './layer-factory';
+import {Field, KeplerTable} from 'utils';
+import {DataContainerInterface} from 'utils/table-utils/data-container-interface';
+import {GpuFilter} from 'utils/table-utils/kepler-table';
+
+export {LAYER_VIS_CONFIGS} from './layer-factory';
+
+export type LayerColumn = {value: string | null; fieldIdx: number; optional?: boolean};
+
+export type LayerColumns = {
+  [key: string]: LayerColumn;
+};
+export type VisualChannelDomain = number[] | string[];
+export type VisualChannelField = Field | null;
+
+export type LayerConfig = {
```

Comment:
let's rename this constant from `LayerConfig` to `LayerBaseConfig`. And remove all the `VisualChannel` properties (`colorField`, `colorDomain`, `colorScale`, `heightField`, `heightDomain`, `heightScale`, `sizeDomain`, `sizeScale`, `sizeField`)

Visual channels various per layer, some layer have colorField and strokeColorField, and radiusField, some doesn't have any channels. We can add individual visual channel configs in each layer type. E.g. Point layer can be:


```
type PointVisualChannelConfig = {
  colorField: VisualChannelField;
  colorScale: VisualChannelScale;
  colorDomain: VisualChannelDomain;
  strokeColorField: VisualChannelField;
  strokeColorScale:  VisualChannelScale;
  strokeColorDomain: VisualChannelDomain;
  sizeField: VisualChannelField;
  sizeScale: VisualChannelScale;
  sizeDomain: VisualChannelDomain;
}
type PointLayer = BaseLayerConfig & PointVisualChannelConfig
 ```